src/components/Carousel/Carousel.js

```diff
@@ -47,61 +44,85 @@ class Carousel extends Component {
+  getSlidesCount = () => {
+    const sizeValues = {
+      desktop: 3,
+      tablet: 2,
+      mobile: 1
+    };
+
+    let count;
+
+    if (window.innerWidth >= 1024) {
+      count = sizeValues.desktop;
+    } else if (window.innerWidth >= 768) {
+      count = sizeValues.tablet;
+    } else if (window.innerWidth >= 320) {
+      count = sizeValues.mobile;
     }
 
-    this.carouselViewport.current.parentElement.style.width = carouselWidth;
+    return count;
+  };
+
+  resizeTheCarousel() {
+    this.setState({ slidesCount: this.getSlidesCount() });
+    this.countTheSlideWidth();
+  }
+
+  countTheSlideWidth() {
+    const { scrollWidth } = this.carouselViewport.current.parentElement;
+    const { slidesCount } = this.state;
+
+    this.setState({ widthOfSlide: scrollWidth / slidesCount });
   }
 
   checkIfSlidesAllTheWayOver() {
-    const { scrollLeft, clientWidth, scrollWidth } = this.carouselViewport.current;
+    const {
+      scrollLeft,
+      clientWidth,
+      scrollWidth
+    } = this.carouselViewport.current;
     // if scrollLeft == 0
     // hide left button
-    const allTHeWayLeftValue = (scrollLeft === 0);
+    const allTHeWayLeftValue = scrollLeft === 0;
 
     // if scrollLeft + viewPortOffset.length === whole viewPort length
     // 9 cards - each 120px: 9 * 210all === whole viewPort length
     // hide the rightScrollButton
-    const allTheWayRightValue = ((scrollLeft + clientWidth) === scrollWidth);
+    const allTheWayRightValue = scrollLeft + clientWidth === scrollWidth;
     const { allTheWayLeft, allTheWayRight } = this.state;
 
-    (allTheWayLeft !== allTHeWayLeftValue || allTheWayRight !== allTheWayRightValue)
+    (allTheWayLeft !== allTHeWayLeftValue
+      || allTheWayRight !== allTheWayRightValue)
       && this.setState({
         allTheWayLeft: allTHeWayLeftValue,
         allTheWayRight: allTheWayRightValue
       });
   }
 
   checkNumOfSlidesToScroll() {
-    const numOfSlidesToScroll = window.innerWidth <= 900 ? 2 : 4;
+    const numOfSlidesToScroll = window.innerWidth <= 900 ? 1 : 2;
     const { numOfSlidesToScroll: numOfSlides } = this.state;
 
-    numOfSlides !== numOfSlidesToScroll && this.setState({ numOfSlidesToScroll });
+    numOfSlides !== numOfSlidesToScroll
+      && this.setState({ numOfSlidesToScroll });
   }
 
   handleClick(e) {
-    const clickedBtn = (e.currentTarget.classList.contains(`${CN}__left-nav`)) ? 'left' : 'right';
+    const clickedBtn = e.currentTarget.classList.contains(`${CN}__left-nav`)
+      ? 'left'
+      : 'right';
     const { numOfSlidesToScroll, widthOfSlide } = this.state;
     const carouselViewport = this.carouselViewport.current;
     const step = numOfSlidesToScroll * widthOfSlide;
     const newPos = clickedBtn === 'left'
-      ? (carouselViewport.scrollLeft - (step))
-      : (carouselViewport.scrollLeft + (step));
+      ? carouselViewport.scrollLeft - step
+      : carouselViewport.scrollLeft + step;
     const timeToMoveOneSlide = 200;
```

Comment:
this is 200px on any kind of resolutions?

Reply:
looks like this is 200 miliseconds

Reply:
need some sleep